src/commands/content-item/tree.ts

```diff
@@ -0,0 +1,243 @@
+import { getDefaultLogPath } from '../../common/log-helpers';
+import { Argv, Arguments } from 'yargs';
+import { join, extname, resolve } from 'path';
+import { ConfigurationParameters } from '../configure';
+import { lstat, readdir, readFile } from 'fs';
+import { promisify } from 'util';
+
+import { ContentItem, ContentRepository } from 'dc-management-sdk-js';
+import {
+  ContentDependancyTree,
+  ItemContentDependancies,
+  RepositoryContentItem
+} from '../../common/content-item/content-dependancy-tree';
+import { ContentMapping } from '../../common/content-item/content-mapping';
+
+export const command = 'tree <dir>';
+
+export const desc = 'Print a content dependency tree from content in the given folder.';
+
+export const LOG_FILENAME = (platform: string = process.platform): string =>
+  getDefaultLogPath('item', 'tree', platform);
+
+export const builder = (yargs: Argv): void => {
+  yargs.positional('dir', {
+    type: 'string',
+    describe: 'Path to the content items to build a tree from. Should be in the same format as an export.'
+  });
+};
+
+interface TreeOptions {
+  dir: string;
+}
+
+export const traverseRecursive = async (path: string, action: (path: string) => Promise<void>): Promise<void> => {
+  const dir = await promisify(readdir)(path);
+
+  dir.sort();
+
+  for (let i = 0; i < dir.length; i++) {
+    let contained = dir[i];
+    contained = join(path, contained);
+    const stat = await promisify(lstat)(contained);
+    if (stat.isDirectory()) {
+      await traverseRecursive(contained, action);
+    } else {
+      await action(contained);
+    }
+  }
+};
+
+export const prepareContentForTree = async (
+  repo: { basePath: string; repo: ContentRepository },
+  argv: Arguments<TreeOptions & ConfigurationParameters>
+): Promise<ContentDependancyTree> => {
+  const contentItems: RepositoryContentItem[] = [];
+  const schemaNames = new Set<string>();
+
+  await traverseRecursive(resolve(repo.basePath), async path => {
+    // Is this valid content? Must have extension .json to be considered, for a start.
+    if (extname(path) !== '.json') {
+      return;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let contentJSON: any;
+    try {
+      const contentText = await promisify(readFile)(path, { encoding: 'utf8' });
+      contentJSON = JSON.parse(contentText);
+    } catch (e) {
+      console.error(`Couldn't read content item at '${path}': ${e.toString()}`);
+      return;
+    }
+
+    schemaNames.add(contentJSON.body._meta.schema);
+
+    contentItems.push({ repo: repo.repo, content: new ContentItem(contentJSON) });
+  });
+
+  return new ContentDependancyTree(contentItems, new ContentMapping());
+};
+
+type CircularLink = [number, number];
```

Comment:
Maybe change the circularLinks to be an arrary of an object, as it's not clear what each _number_ represents.

Reply:
Each number is an index in the lines of the tree. Neither of the numbers have any special meaning - they mean the same thing - it's just a pair of two linked indices. I think giving them names like `index1`, `index2` might not be great.

Reply:
```
type CircularLinkFrom = number;
type CircularLinkTo = number;
type CircularLink = [CircularLinkFrom, CircularLinkTo];
```